src/index.ts

```diff
@@ -42,6 +42,9 @@ async function activateCellVim(
     },
     isToggled: () => enabled
   });
+
+  const userKeybinds = (await settingRegistry.get(`${PLUGIN_NAME}:plugin`, 'extraKeybindings')).composite as unknown as Array<Keybinding>
```

Comment:
1. Maybe for consistency:

```suggestion
  const userKeybindings = (await settingRegistry.get(`${PLUGIN_NAME}:plugin`, 'extraKeybindings')).composite as unknown as Array<Keybinding>
```

2. There does not seem to be a listener to update the keybindings if those change. You should ideally add it, or less preferably instruct the user to reload JupyterLab after changing settings

Reply:
> There does not seem to be a listener to update the keybindings if those change.

[There is already an event listener](https://github.com/peytondmurray/jupyterlab-vim/blob/085ebdb8cea9ded13309de4992a36ad7370cea07/src/index.ts#L120) that triggers when settings change, so I've added a few lines to set the `userKeybindings` in there.

But there's a separate bug that prevents this from working: inside the function that gets called when the settings change, there's a [line which re-registers commands to the `CommandRegistry`](https://github.com/peytondmurray/jupyterlab-vim/blob/085ebdb8cea9ded13309de4992a36ad7370cea07/src/index.ts#L92). Any time this is called (i.e. any time after the initial page load) it tries to register commands that have _already_ been registered, [resulting in an error](https://lumino.readthedocs.io/en/1.x/api/commands/classes/commandregistry.html#addcommand). What is the right way to handle this situation?

If I were writing this extension, naively I would just to register commands for this extension once inside `activateCellVim`, not in `updateSettings`; then, I'd just enable or disable the registered commands based on whether the extension is enabled or disabled. Anyway, it doesn't look like that's what is happening here; it's difficult for me to understand the original intent behind the current approach, so I'm reluctant to just change it. What are your thoughts?

Reply:
I agree that usually registering commands once is a better choice. There is a way to dispose already registered command - when you call `addCommand()` you get a `DisposableDelegate` so you can remove the command later if you store a reference to it (but currently the result of every `addCommand()` is ignored in `addJLabCommands()` rather than stored.

Reply:
Okay, I've just removed the part that registers/disposes of the commands, it doesn't seem to do anything here. Thanks for the advice.